### ISSUE TITLE
Copy curves and re-use existing ones

### ISSUE THREAD

**Issue by LeXXik** [Pomax/bezierjs#123]
I am using it with Playcanvas WebGL game engine for path-finding and whatnot.  Here is a use case I often meet:

I find two sets of coordinates, which are stored in Vector3 objects. Using these coordinates, I create 2 curves A and B. After some time, curve A needs to take place of curve B, while curve B needs to use a new set of coordinates.

1. Currently, I copy vector coordinates from B to A. However, I have to destroy the old A curve and generate a new A one via `new Bezier()` with the updated vector coordinates. Copying would have been better for performance reasons. Is there a way to copy curve values from another curve without creating a new object?

2. Now, curve B got a new set of coordinates, and I wish I could set the current B curve to them. Instead, again, I destroy the old B curve and create a new one via `new Bezier()`, using the new coordinates. Is there a way to re-use existing curve object?

Thank you in advance! Love the library!

**Comment by Pomax** [comment]
You could copy the internal `Bezier.points` from B to A, then _make sure to call A.update()_, and then make `B.points` a new array of points, _and call B.update() too_

(without the `update()` call, none of the cached and precomputed values will get updated and your curve will basically be doing entirely the wrong thing in certain cases).

That said I'm honestly not sure whether this makes sense: building a new curve isn't very intensive/expensive, did you profile your code and it turned out the hotspot was _actually_ the `new Bezier(...)` call? O_o

**Comment by LeXXik** [comment]
Works like a charm! :) Thank you, @Pomax!
Yes, I oversimplified the example use case to make it easy to understand the question. For the use case I referred to, there would be no need in copying the values. Its a trivial case and new memory allocated with `new` keyword would be so small, it would make no difference. However, in my case I have a pool service that handles a large number of objects, including the curves. Instantiating a big enough number can kick in the garbage collector which may introduce a spike. Re-using the objects alleviates it.
The issue can be closed now!
